File: csharp/Snake/Snake.cs
namespace Snake
{
    public partial class Snake
    {
        // povolene smery pohybu hada
        public enum Direction
        {
            Right,
            Left,
            Up,
            Down,
        }

        // nedostane data hry, vraci jmeno hada
        public string? Index()
        {
            return Name;
        }

        // dostane data hry, vraci prazdnou odpoved
        public string Init(Game game)
        {
            return string.Empty;
        }

        // dostane data hry, vraci smer pohybu hada
        public Direction Move(Game game)
        {
            /*
                SEM UMISTUJTE SVUJ KOD
            */

            // pro ukazku se vraci nahodny smer
            Direction[] directions = (Direction[])Enum.GetValues(typeof(Direction));
            Direction randomDirection = directions[(new Random()).Next(directions.Length)];

            return randomDirection;
        }
    }
}

File: javascript/app/snake.js
// identifikator a jmeno hada z konfiguracniho souboru .env
const { id, name } = require('../conf');
// povolene smery pohybu hada
const directions = ["Up", "Down", "Left", "Right"];

// nedostane data hry, vraci jmeno hada
exports.index = () => {
	return name;
}

// dostane data hry, vraci prazdnou odpoved
exports.init = (game) => {
	return '';
}

// dostane data hry, vraci smer pohybu hada
exports.move = (game) => {
	/*
		SEM UMISTUJTE SVUJ KOD
	*/
	
	// pro ukazku se vraci nahodny smer
	return {"direction": directions[Math.floor(Math.random() * 4)]};
}
